engines/made/metaengine.cpp

```diff
@@ -89,6 +95,91 @@ Common::Error MadeMetaEngine::createInstance(OSystem *syst, Engine **engine, con
 	return Common::kNoError;
 }
 
+Common::KeymapArray MadeMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace Made;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "made-main", "MADE main");
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game Keymappings"));
+
+	Action *act;
+
+	act = new Action(kStandardActionLeftClick, _("Left Click"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	act->addDefaultInputMapping("KP_PLUS");
+	engineKeyMap->addAction(act);
+
+	act = new Action(kStandardActionRightClick, _("Right Click"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	act->addDefaultInputMapping("KP_MINUS");
+	engineKeyMap->addAction(act);
+
+	act = new Action(kStandardActionSkip, _("Skip"));
+	act->setKeyEvent(KeyState(KEYCODE_ESCAPE, ASCII_ESCAPE));
+	act->addDefaultInputMapping("ESCAPE");
+	act->addDefaultInputMapping("JOY_Y");
+	act->allowKbdRepeats();
+	engineKeyMap->addAction(act);
+
+	act = new Action("CRSORUP", _("Cursor Up"));
+	act->setCustomEngineActionEvent(kActionCursorUp);
+	act->addDefaultInputMapping("UP");
+	act->addDefaultInputMapping("KP8");
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSORDOWN", _("Cursor Down"));
+	act->setCustomEngineActionEvent(kActionCursorDown);
+	act->addDefaultInputMapping("DOWN");
+	act->addDefaultInputMapping("KP2");
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSORLEFT", _("Cursor Left"));
+	act->setCustomEngineActionEvent(kActionCursorLeft);
+	act->addDefaultInputMapping("LEFT");
+	act->addDefaultInputMapping("KP4");
+	gameKeyMap->addAction(act);
+
+	act = new Action("CRSORRIGHT", _("Cursor Right"));
+	act->setCustomEngineActionEvent(kActionCursorRight);
+	act->addDefaultInputMapping("RIGHT");
+	act->addDefaultInputMapping("KP6");
+	gameKeyMap->addAction(act);
+
+	act = new Action("MENU", _("Menu"));
+	act->setCustomEngineActionEvent(kActionMenu);
+	act->addDefaultInputMapping("F1");
+	act->addDefaultInputMapping("JOY_UP"); 
```

Comment:
It would probably be a good idea to map these to the remaining A/B/X/Y/L/R buttons before resorting to the D-pad so they're still accessible with joysticks that lack analog sticks.